sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/PassThroughThenCleanup.java

```diff
@@ -79,6 +81,6 @@ public void processElement(ProcessContext c) {
   }
 
   abstract static class CleanupOperation implements Serializable {
-    abstract void cleanup(PipelineOptions options) throws Exception;
+    abstract void cleanup(DoFn.ProcessContext options) throws Exception;
```

Comment:
Yes, this seems reasonable, assuming job id is the only thing to which you want CleanupOperation to have access (rather than arbitrary side inputs).